salt/beacons/__init__.py

```diff
@@ -95,6 +97,17 @@ def process(self, config, grains):
                         continue
                 # Update __grains__ on the beacon
                 self.beacons[fun_str].__globals__['__grains__'] = grains
+
+                # Run the validate function if it's available,
+                # otherwise there is a warning about it being missing
+                if validate_str in self.beacons:
+                    valid, vcomment = self.beacons[validate_str](b_config[mod])
+
+                    if not valid:
+                        log.info('Beacon {0} configuration invalid, '
```

Comment:
The validate function was originally `validate` and was only used by the beacon execution module when adding or modifying beacons.  It was updated to `__validate__` with the idea that it would be used in loader.py when the beacon modules were initially loaded.  Unfortunately at that point the configuration for the beacon wasn't available so the call to validate wouldn't work.  So this change renames it back to validate and moves the call into beacons/__init__.py instead of the loader.py

Regarding string replacement, are you suggesting that I use:
```log.info('Beacon %s configuration is invalid....``` instead of format?  We use the format method all over, including in calls to logging.